zerver/actions/message_send.py

```diff
@@ -1327,6 +1331,32 @@ def check_private_message_policy(
         raise JsonableError(_("Direct messages are disabled in this organization."))
 
 
+def check_direct_message_policy(
+    realm: Realm, sender: UserProfile, user_profiles: Sequence[UserProfile]
+) -> None:
+    
+    if sender.is_bot or (len(user_profiles) == 1 and user_profiles[0].is_bot):
+        # We allow PMs only between users and bots, to avoid
+        # breaking the tutorial as well as automated
+        # notifications from system bots to users.
+        return
+    
+    direct_message_permission_group = access_user_group_by_id(realm.direct_message_permission_group_id.id, sender, for_read=True)
+    direct_message_initiator_group = access_user_group_by_id(realm.direct_message_initiator_group_id.id, sender, for_read=True)
+
+    if is_user_in_group(direct_message_initiator_group, sender) == False:
+        raise JsonableError(_("You do not have permission to send message in this organization"))
```

Comment:
I think this check is wrong -- for "initiator", we need to query whether at least one message exists in the conversation, and if it does